pkgs/development/libraries/libxml2/default.nix

```
@@ -51,6 +52,23 @@ stdenv.mkDerivation (finalAttrs: {
     hash = "sha256-J3KUyzMRmrcbK8gfL0Rem8lDW4k60VuyzSsOhZoO6Eo=";
   };
 
+  patches = [
+    # same as upstream patch but fixed conflict and added required import:
+    # https://gitlab.gnome.org/GNOME/libxml2/-/commit/acbbeef9f5dcdcc901c5f3fa14d583ef8cfd22f0.diff
+    ./CVE-2025-6021.patch
+    (fetchpatch2 {
+      name = "CVE-2025-49794-49796.patch";
+      url = "https://gitlab.gnome.org/GNOME/libxml2/-/commit/f7ebc65f05bffded58d1e1b2138eb124c2e44f21.patch";
+      hash = "sha256-k+IGq6pbv9EA7o+uDocEAUqIammEjLj27Z+2RF5EMrs=";
+    })
+    # same as upstream but skipped the tests, they were rewritten from schematron to C
+    # https://gitlab.gnome.org/GNOME/libxml2/-/commit/c24909ba2601848825b49a60f988222da3019667.patch
+    ./CVE-2025-49795.patch
```

```suggestion
    (fetchpatch2 {
      name = "CVE-2025-49795.patch";
      url = "https://gitlab.gnome.org/GNOME/libxml2/-/commit/c24909ba2601848825b49a60f988222da3019667.patch";
      hash = "sha256-r7PYKr5cDDNNMtM3ogNLsucPFTwP/uoC7McijyLl4kU=";
      excludes = [ "runtest.c" ]; # tests were rewritten in C and are on schematron for 2.13.x, meaning this does not apply
    })
```
I suggest fetching this patch too, makes review simpler because there is less to compare. And delete the vendored patch file then.

---

Thanks, good to know that we can exclude files from the patch

(Probably I should resulve these from the other PR because commit hashes will change and the cherry-pick check will fail)

---

Honestly, the split PR is quite different because it acts on the split package. I wouldn't worry too much about the cherry-pick check